js/encoding/tlv-0_2-wire-format.js

```diff
@@ -112,79 +112,7 @@ Tlv0_2WireFormat.prototype.encodeInterest = function(interest)
-    // Tlv0_2WireFormat.
-    return Tlv0_2WireFormat.encodeInterestV03_
-      (interest, signedPortionBeginOffset, signedPortionEndOffset);
-
-  var encoder = new TlvEncoder(256);
-  var saveLength = encoder.getLength();
-
-  // Encode backwards.
-  if (interest.getForwardingHint().size() > 0) {
-    if (interest.getSelectedDelegationIndex() != null)
-      throw new Error
-        ("An Interest may not have a selected delegation when encoding a forwarding hint");
-    if (interest.hasLink())
-      throw new Error
-        ("An Interest may not have a link object when encoding a forwarding hint");
-
-    var forwardingHintSaveLength = encoder.getLength();
-    Tlv0_2WireFormat.encodeDelegationSet_(interest.getForwardingHint(), encoder);
-    encoder.writeTypeAndLength(
-      Tlv.ForwardingHint, encoder.getLength() - forwardingHintSaveLength);
-  }
-
-  encoder.writeOptionalNonNegativeIntegerTlv
-    (Tlv.SelectedDelegation, interest.getSelectedDelegationIndex());
-  var linkWireEncoding = interest.getLinkWireEncoding(this);
-  if (!linkWireEncoding.isNull())
-    // Encode the entire link as is.
-    encoder.writeBuffer(linkWireEncoding.buf());
-
-  encoder.writeOptionalNonNegativeIntegerTlv
-    (Tlv.InterestLifetime, interest.getInterestLifetimeMilliseconds());
-
-  // Encode the Nonce as 4 bytes.
-  if (interest.getNonce().isNull() || interest.getNonce().size() == 0)
-    // This is the most common case. Generate a nonce.
-    encoder.writeBlobTlv(Tlv.Nonce, Crypto.randomBytes(4));
-  else if (interest.getNonce().size() < 4) {
-    var nonce = Buffer(4);
-    // Copy existing nonce bytes.
-    interest.getNonce().buf().copy(nonce);
-
-    // Generate random bytes for remaining bytes in the nonce.
-    for (var i = interest.getNonce().size(); i < 4; ++i)
-      nonce[i] = Crypto.randomBytes(1)[0];
-
-    encoder.writeBlobTlv(Tlv.Nonce, nonce);
-  }
-  else if (interest.getNonce().size() == 4)
-    // Use the nonce as-is.
-    encoder.writeBlobTlv(Tlv.Nonce, interest.getNonce().buf());
-  else
-    // Truncate.
-    encoder.writeBlobTlv(Tlv.Nonce, interest.getNonce().buf().slice(0, 4));
-
-  Tlv0_2WireFormat.encodeSelectors(interest, encoder);
-  var tempOffsets = Tlv0_2WireFormat.encodeName(interest.getName(), encoder);
-  var signedPortionBeginOffsetFromBack =
-    encoder.getLength() - tempOffsets.signedPortionBeginOffset;
-  var signedPortionEndOffsetFromBack =
-    encoder.getLength() - tempOffsets.signedPortionEndOffset;
-
-  encoder.writeTypeAndLength(Tlv.Interest, encoder.getLength() - saveLength);
-  var signedPortionBeginOffset =
-    encoder.getLength() - signedPortionBeginOffsetFromBack;
-  var signedPortionEndOffset =
-    encoder.getLength() - signedPortionEndOffsetFromBack;
-
-  return { encoding: new Blob(encoder.getOutput(), false),
-           signedPortionBeginOffset: signedPortionBeginOffset,
-           signedPortionEndOffset: signedPortionEndOffset };
+  return Tlv0_2WireFormat.encodeInterestV03_(interest);
```

Comment:
Looks good to me. We have to tag the latest commit as a release before committing this change, though.

Reply:
why?

Reply:
It is a good idea to have a disjoint point between v.02 and v.03. After this commit, all packets will be encoded into v0.3.

Reply:
There were no noteworthy changes since the last release (v0.19), just a couple of additional debug messages. Making another release now seems quite overkill to me.

Reply:
How else can we mark the current commit for future references? If an application needs to use `v.02` it should be easy to find the current commit.

Reply:
you can just use v0.19, the changelog will say that v0.20 changed the default encoding.

Reply:
Wait, the latest version (`v.019`) is very recent. I somehow was thinking about `v.018`. We do not need to mark the current commit then.